core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -74,6 +75,25 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
   def recoverWith[A](fa: F[A])(pf: PartialFunction[E, F[A]]): F[A] =
     handleErrorWith(fa)(e =>
       pf applyOrElse(e, raiseError))
+  /**
+   * Often E is Throwable. Here we try to call pure or catch
+   * and raise.
+   */
+  def tryCatch[A](a: => A)(implicit ev: Throwable =:= E): F[A] =
```

Comment:
What do you think about naming this `catchNonFatal` to match the method on `Xor`?